<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1362</width>
    <height>713</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="imgLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>471</width>
      <height>521</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="imgLabel2">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>30</y>
      <width>491</width>
      <height>521</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QDial" name="dial">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>-7</y>
      <width>141</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>20</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotation</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>90</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>5</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
    <property name="sliderPosition">
     <number>5</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>90</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Gamma</string>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>10</y>
      <width>20</width>
      <height>631</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSlider" name="size_Img">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>60</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
    <property name="sliderPosition">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>60</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Size</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>120</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Gaussian</string>
    </property>
   </widget>
   <widget class="QSlider" name="gaussian_QSlider">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>120</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>4</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="erosion">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>170</y>
      <width>201</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5</number>
    </property>
    <property name="maximum">
     <number>5</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>150</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Dilate_Erosion</string>
    </property>
   </widget>
   <widget class="QLabel" name="log">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>200</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Log</string>
    </property>
   </widget>
   <widget class="QSlider" name="Qlog">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>200</y>
      <width>141</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>4</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
    <property name="sliderPosition">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QCheckBox" name="sobel">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>260</y>
      <width>171</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Sobel Derivative</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="sobel_x">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>290</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>X-axis</string>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>310</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QCheckBox" name="sobel_y">
    <property name="geometry">
     <rect>
      <x>1230</x>
      <y>290</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Y-axis</string>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>240</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QCheckBox" name="gray">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>330</y>
      <width>97</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Grayscale</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="hist">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>360</y>
      <width>161</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>histogram Equalization</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cbLog">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>390</y>
      <width>161</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Log</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cbLap">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>420</y>
      <width>161</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Laplacian</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>495</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Min</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>525</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Max</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="canny">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>465</y>
      <width>171</width>
      <height>22</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>Canny Edge Detector</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QSlider" name="canny_min">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>490</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="canny_max">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>520</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>400</number>
    </property>
    <property name="value">
     <number>200</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>450</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>590</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Hoàng Huy - Hiếu Bốn</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>570</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
    <property name="text">
     <string>RESET</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1362</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionPrint"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionSmall"/>
    <addaction name="actionBig"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_6">
    <property name="title">
     <string>Chương 2</string>
    </property>
    <addaction name="actionRotation"/>
    <addaction name="actionAffine"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_9">
    <property name="title">
     <string>Chương 3</string>
    </property>
    <addaction name="actioAnhXam"/>
    <addaction name="actionNegative"/>
    <addaction name="actionLog"/>
    <addaction name="actionHistogram"/>
    <addaction name="actionGamma"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_10">
    <property name="title">
     <string>Chương 4</string>
    </property>
    <addaction name="actionFilter_Average"/>
    <addaction name="actionGaussan"/>
    <addaction name="actionHigh_Boost"/>
    <addaction name="actionLaplacian"/>
    <addaction name="actionUnsharp"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_11">
    <property name="title">
     <string>Chương 5</string>
    </property>
    <widget class="QMenu" name="menuFourier_2">
     <property name="title">
      <string>Fourier</string>
     </property>
     <addaction name="actionTanSo"/>
     <addaction name="separator"/>
     <addaction name="actionIdeal_LPF"/>
     <addaction name="actionButter_LPF"/>
     <addaction name="actionGaussian_LPF"/>
     <addaction name="separator"/>
     <addaction name="actionIdeal_HPF"/>
     <addaction name="actionButterworth_HPF"/>
     <addaction name="actionGaussian_HPF"/>
    </widget>
    <widget class="QMenu" name="menuType_Here">
     <property name="title">
      <string>Type Here</string>
     </property>
    </widget>
    <addaction name="menuFourier_2"/>
    <addaction name="menuType_Here"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_12">
    <property name="title">
     <string>Chương 7</string>
    </property>
    <addaction name="actionDilate"/>
    <addaction name="actionErode"/>
    <addaction name="actionOpen_2"/>
    <addaction name="actionClose"/>
    <addaction name="actionHit_miss"/>
    <addaction name="actionMorboundary"/>
    <addaction name="actionGradient"/>
    <addaction name="actionConvex"/>
    <addaction name="actionCanny"/>
   </widget>
   <widget class="QMenu" name="menuCh_ng_13">
    <property name="title">
     <string>Chương 8</string>
    </property>
    <addaction name="actionx_direcction_Sobel"/>
    <addaction name="actiony_direction_Sobel"/>
    <addaction name="actionLaplacian_2"/>
    <addaction name="actionLaplacian_of_Gaussian"/>
    <addaction name="actionHough"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionQt"/>
    <addaction name="actionAuthor"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCh_ng_6"/>
   <addaction name="menuCh_ng_9"/>
   <addaction name="menuCh_ng_10"/>
   <addaction name="menuCh_ng_11"/>
   <addaction name="menuCh_ng_12"/>
   <addaction name="menuCh_ng_13"/>
   <addaction name="menuView"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionPrint"/>
   <addaction name="separator"/>
   <addaction name="actionQuit"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset>
     <normaloff>../../bong.jpg</normaloff>../../bong.jpg</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>../../.designer/backup/open.png</normaloff>../../.designer/backup/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>../../.designer/backup/Save.png</normaloff>../../.designer/backup/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset>
     <normaloff>../../.designer/backup/print.png</normaloff>../../.designer/backup/print.png</iconset>
   </property>
   <property name="text">
    <string>Print</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset>
     <normaloff>../../.designer/backup/SaveAs.png</normaloff>../../.designer/backup/SaveAs.png</iconset>
   </property>
   <property name="text">
    <string>Save as...</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset>
     <normaloff>../../.designer/backup/Quit.png</normaloff>../../.designer/backup/Quit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actioAnhXam">
   <property name="text">
    <string>Graycsale</string>
   </property>
  </action>
  <action name="actionCanny">
   <property name="text">
    <string>Canny</string>
   </property>
  </action>
  <action name="actionNegative">
   <property name="text">
    <string>Negative</string>
   </property>
  </action>
  <action name="actionHistogram">
   <property name="text">
    <string>Histogram</string>
   </property>
  </action>
  <action name="actionLog">
   <property name="text">
    <string>Log </string>
   </property>
  </action>
  <action name="actionGaussan">
   <property name="text">
    <string>Gaussan</string>
   </property>
  </action>
  <action name="actionHigh_Boost">
   <property name="text">
    <string>High Boost</string>
   </property>
  </action>
  <action name="actionLaplacian">
   <property name="text">
    <string>Laplacian</string>
   </property>
  </action>
  <action name="actionFilter_Average">
   <property name="text">
    <string>Filter Average</string>
   </property>
  </action>
  <action name="actionUnsharp">
   <property name="text">
    <string>Unsharp</string>
   </property>
  </action>
  <action name="actionCh_ng_5">
   <property name="text">
    <string>Chương 5</string>
   </property>
  </action>
  <action name="actionTanSo">
   <property name="text">
    <string>Ảnh Miền Tần Số</string>
   </property>
  </action>
  <action name="actionIdeal_LPF">
   <property name="text">
    <string>Ideal LPF</string>
   </property>
  </action>
  <action name="actionButter_LPF">
   <property name="text">
    <string>Butterworth LPF</string>
   </property>
  </action>
  <action name="actionGaussian_LPF">
   <property name="text">
    <string>Gaussian LPF</string>
   </property>
  </action>
  <action name="actionIdeal_HPF">
   <property name="text">
    <string>Ideal HPF</string>
   </property>
  </action>
  <action name="actionButterworth_HPF">
   <property name="text">
    <string>Butterworth HPF</string>
   </property>
  </action>
  <action name="actionGaussian_HPF">
   <property name="text">
    <string>Gaussian HPF</string>
   </property>
  </action>
  <action name="actiondilate">
   <property name="text">
    <string>dilate</string>
   </property>
  </action>
  <action name="actionErode">
   <property name="text">
    <string>Erode</string>
   </property>
  </action>
  <action name="actionOpen_2">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionHit_miss">
   <property name="text">
    <string>Hit-miss</string>
   </property>
  </action>
  <action name="actionDilate">
   <property name="text">
    <string>Dilate</string>
   </property>
  </action>
  <action name="actionMorboundary">
   <property name="text">
    <string>Morboundary</string>
   </property>
  </action>
  <action name="actionGradient">
   <property name="text">
    <string>Gradient</string>
   </property>
  </action>
  <action name="actionConvex">
   <property name="text">
    <string>Convex</string>
   </property>
  </action>
  <action name="actionx_direcction_Sobel">
   <property name="text">
    <string>x_direcction Sobel</string>
   </property>
  </action>
  <action name="actiony_direction_Sobel">
   <property name="text">
    <string>y_direction Sobel</string>
   </property>
  </action>
  <action name="actionLaplacian_2">
   <property name="text">
    <string>Sobel Laplacian</string>
   </property>
  </action>
  <action name="actionLaplacian_of_Gaussian">
   <property name="text">
    <string>Laplacian of Gaussian</string>
   </property>
  </action>
  <action name="actionHough">
   <property name="text">
    <string>Hough</string>
   </property>
  </action>
  <action name="actionSmall">
   <property name="text">
    <string>Zoom in</string>
   </property>
  </action>
  <action name="actionRotation">
   <property name="text">
    <string>Rotation</string>
   </property>
  </action>
  <action name="actionAffine">
   <property name="text">
    <string>Affine</string>
   </property>
  </action>
  <action name="actionGamma">
   <property name="text">
    <string>Gamma</string>
   </property>
  </action>
  <action name="actionBig">
   <property name="text">
    <string>Zoom out</string>
   </property>
  </action>
  <action name="actionQt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionAuthor">
   <property name="text">
    <string>Author</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
